src/librustdoc/html/static/js/main.js

```diff
@@ -979,6 +979,121 @@ function hideThemeButtonState() {
     onHashChange(null);
     window.addEventListener("hashchange", onHashChange);
     searchState.setup();
+
+    /////// --scrape-examples interactions
+
+    // Scroll code block to put the given code location in the middle of the viewer
+    function scrollToLoc(elt, loc) {
```

Comment:
Does this function not already exists? I'm pretty sure go-to-definition uses something like it, can you reuse whatever it's doing?

Reply:
Go-to-definition just uses standard hyperlinks with #-based line numbers. This works for the standalone code-viewing page, but won't work for our use case since the element is not scrollable by default.

Reply:
It might be nice to add an anchor for each example ... but that doesn't need to go in this PR.